### PR TITLE
Create task 0 like other tasks

### PR DESCRIPTION
**Pull request by SamTebbs33**
This PR makes the creation of task 0 the same as normal tasks, to make the code cleaner.

### CODE REVIEW

**Comment by SamTebbs33** [comment]
@DrDeano ping a ding ding

File: src/kernel/scheduler.zig [hunk 1]
```diff
@@ -138,14 +138,12 @@ pub fn init(allocator: Allocator, mem_profile: *const mem.MemProfile) Allocator.
     tasks = TailQueue(*Task){};
 
     // Set up the init task to continue execution
-    current_task = try allocator.create(Task);
+    current_task = try Task.create(0, true, &vmm.kernel_vmm, allocator);
     errdefer allocator.destroy(current_task);
-    // PID 0
-    current_task.pid = 0;
+
+    allocator.free(current_task.kernel_stack);
```
Comment (DrDeano) [inline src/kernel/scheduler.zig]: Might be worth adding a comment for why this free is here as a new kernel stack is allocated in `create`. Or have a way of not allocating a kernel stack in `Task.create`. Could have a `createWithStack` and you pass in address range for the kernel stack and a `createWithoutStack` which allocates the kernel stack. The naming of the functions can something else if you like. As allocating then immediately freeing memory does seem to be a bit dumb.

Comment (SamTebbs33) [reply src/kernel/scheduler.zig]: Yeah I definitely agree. Perhaps a boolean parameter would be a nice idea.

**Comment by SamTebbs33** [comment]
I've finally finished this! I was stuck on a page fault and index out of bounds error for a little while.

**Pull request merged**